contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,415 @@
+                  new String(tuple.get(afi.fieldInfo.getColumnName()).toString()));
+              break;
+
+            case LONG:
+              ((PojoUtils.SetterLong<Object>)afi.setterOrGetter).set(newObj,
+                  (long)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            default:
+              ((PojoUtils.Setter<Object, byte[]>)afi.setterOrGetter).set(newObj,
+                  serialize(tuple.get(afi.fieldInfo.getColumnName())));
+              break;
+          }
+        } catch (AvroRuntimeException | IOException e) {
+          LOG.error("Exception in setting value" + e.getMessage());
+          fieldErrorCnt++;
+        }
+
+      }
+    } catch (Exception ex) {
+      LOG.error("Generic Exception in setting value" + ex.getMessage());
+      errorCnt++;
+      newObj = null;
+    }
+    return newObj;
+  }
+
+  /**
+   * Use reflection to generate field info values if the user has not provided
+   * the inputs mapping
+   * 
+   * @return String representing the POJO field to Avro field mapping
+   */
+  public String generateFieldInfoInputs(Class<?> cls)
+  {
+    java.lang.reflect.Field[] fields = cls.getDeclaredFields();
+    StringBuilder sb = new StringBuilder();
+
+    for (int i = 0; i < fields.length; i++) {
+      java.lang.reflect.Field f = fields[i];
+      Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+      sb.append(f.getName() + FIELD_SEPARATOR + f.getName() + FIELD_SEPARATOR + c.getSimpleName().toUpperCase()
+          + RECORD_SEPARATOR);
+    }
+    return sb.substring(0, sb.length() - 1);
+  }
+
+  /**
+   * Creates a map representing fieldName in POJO:field in Generic Record:Data
+   * type
+   * 
+   * @return List of FieldInfo
+   */
+  public List<FieldInfo> createFieldInfoMap(String str)
+  {
+    fieldInfos = new ArrayList<FieldInfo>();
+    StringTokenizer strtok = new StringTokenizer(str, RECORD_SEPARATOR);
+
+    while (strtok.hasMoreTokens()) {
+      String[] token = strtok.nextToken().split(FIELD_SEPARATOR);
+      fieldInfos.add(new FieldInfo(token[0], token[1], SupportType.valueOf(token[2])));
+    }
+    return fieldInfos;
+  }
+
+  @OutputPortFieldAnnotation(schemaRequired = true)
+  public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+    public void setup(PortContext context)
+    {
+      cls = context.getValue(Context.PortContext.TUPLE_CLASS);
+
+      columnFieldSetters = Lists.newArrayList();
+
+      if (getGenericRecordToPOJOFieldsMapping() == null) {
+        setFieldInfos(createFieldInfoMap(generateFieldInfoInputs(getCls())));
+      } else {
+        setFieldInfos(createFieldInfoMap(getGenericRecordToPOJOFieldsMapping()));
+      }
+
+      initColumnFieldSetters(getFieldInfos());
+      initializeActiveFieldSetters();
+    }
+  };
+
+  @Override
+  public void endWindow()
+  {
+    errorCnt = 0;
+    fieldErrorCnt = 0;
+    recordCnt = 0;
+
+  }
+
+  public Class<?> getCls()
```

Comment:
Is it expected by user to set cls? If not, can you please remove this? If yes, can you please give a more intuitive name than cls?